plutus-th/src/Language/Plutus/TH.hs

```diff
@@ -1,19 +1,18 @@
-{-# LANGUAGE TemplateHaskell #-}
-module Language.Plutus.TH (plutus, plutusT, PlcCode, getSerializedCode, getAst, applyPlc) where
+{-# LANGUAGE DataKinds        #-}
+{-# LANGUAGE TemplateHaskell  #-}
+{-# LANGUAGE TypeApplications #-}
+module Language.Plutus.TH (plutus, PlcCode, getSerializedCode, getAst, applyPlc) where
 
 import           Language.Plutus.CoreToPLC.Plugin
 
-import           Language.Haskell.TH
+import qualified Language.Haskell.TH              as TH
 
 -- | Covert a quoted Haskell expression into a corresponding Plutus Core program. Produces an expression of type
 -- 'PlcCode'.
-plutus :: Q Exp -> Q Exp
+plutus :: TH.Q TH.Exp -> TH.Q TH.Exp
 -- We would like to do `addCorePlugin "Language.Plutus.CoreToPLC.Plugin"``, but this needs template-haskell-2.13.0.0,
 -- which is in the next LTS snapshot.
-plutus e = [| plc $(e) |]
-
--- | Covert a (typed) quoted Haskell expression into a corresponding Plutus Core program.
-plutusT :: Q (TExp a) -> Q (TExp PlcCode)
```

Comment:
Will you add `plutusT` again when the issue with `hs-source-exts` has been fixed, or is there another reason to delete it?

Reply:
I don't think you can, here's why. Typed TH basically only works when you lift well-typed Haskell expressions into the typed brackets, because you need to be able to typecheck it *before* TH runtime. But here we are actually generating a new type at TH runtime (the type literal), so there's no way we can typecheck it beforehand. I could use `unsafeTExpCoerce`, but I'd rather not.

I'll add a note in the code.

Reply:
Makes sense. In my mind the biggest downside of not having typed TH is that all expressions are of type `Q Exp` which makes it more difficult to find what you need, esp. in haddocks. Are there any other ones that I am missing?
If not, then we are basically trading a UX feature (typed TH) for another one (named locations). I'm OK with that, especially since we don't use the typed TH stuff at the moment, but it's worth mentioning this trade-off somewhere.

Reply:
I think that's the main one. We can always put it back later with some judicious use of `unsafeTExpCoerce` :see_no_evil: